green/result.py

```diff
@@ -158,15 +161,15 @@ class BaseTestResult:
     I am inherited by ProtoTestResult and GreenTestResult.
     """
 
-    def __init__(self, stream, colors):
-        self.stdout_output = OrderedDict()
-        self.stderr_errput = OrderedDict()
+    def __init__(self, stream, *, colors: Colors | None = None):
+        self.stdout_output: dict[ProtoTest, Any] = {}
+        self.stderr_errput: dict[ProtoTest, Any] = {}
```

Comment:
Insertion order is guaranteed to be preserved in dicts since Python 3.7 (and were implemented as such in 3.6). Since we do not use the extended features of OrderedDict, it is better to just use the normal ones.

https://docs.python.org/3/library/stdtypes.html#mapping-types-dict

> Changed in version 3.7: Dictionary order is guaranteed to be insertion order. This behavior was an implementation detail of CPython from 3.6.